src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,177 @@
+        if (index == 0) {
+            newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        indexNode = findIndexNode(index);
+        newNode = new Node<>(indexNode.prev, value, indexNode);
+        newNode.prev.next = newNode;
+        newNode.next.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexOutOfBoundsException(index);
+        return findIndexNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T oldValue;
+        checkIndexOutOfBoundsException(index);
+        oldValue = findIndexNode(index).value;
+        findIndexNode(index).value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T value;
+        checkIndexOutOfBoundsException(index);
+        if (index == 0) {
+            value = head.value;
+            if (size == 1) {
+                head = null;
+                size--;
+                return value;
+            }
+            head.next.prev = null;
+            head = head.next;
+            size--;
+            return value;
+        } else if (index == size - 1) {
+            value = tail.value;
+            tail.prev.next = null;
+            tail = tail.prev;
+            size--;
+            return value;
+        }
+        Node<T> indexNode = findIndexNode(index);
+        indexNode.prev.next = indexNode.next;
+        indexNode.next.prev = indexNode.prev;
+        size--;
+        return indexNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        if ((head.value == object) || (head.value != null && head.value.equals(object))) {
+            if (size == 1) {
+                head = null;
+                size--;
+                return true;
+            }
+            head.next.prev = null;
+            head = head.next;
+            size--;
+            return true;
+        }
+        Node<T> objectNode = head;
+        for (int i = 0; i < size; i++) {
+            if ((objectNode.value == object)
+                    || (objectNode.value != null && objectNode.value.equals(object))) {
+                objectNode.prev.next = objectNode.next;
+                objectNode.next.prev = objectNode.prev;
+                size--;
+                return true;
+            }
+            objectNode = objectNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        if (size != 0) {
+            return false;
+        }
+        return true;
```

Comment:
Consider simplifying the code. You can directly return the result of the condition instead of using if/else or ternary operation to return true/false. For example:

    // Instead of this
    if (condition) {
        return true;
    } else {
        return false;
    }
    // You can do this
    return condition;

This makes the code cleaner and more straightforward.